src/dune_lang/dune_lang.ml

```diff
@@ -1086,6 +1086,43 @@ module Decoder = struct
     let a, state = a ctx state in
     let b, state = b ctx state in
     ((a, b), state)
+
+  let traverse f l ctx state =
+    Tuple.T2.swap (
+      List.fold_map ~init:state l ~f:(fun state x ->
+        Tuple.T2.swap (f x ctx state)
+      ))
+
+let fields_missing_need_exactly_one loc names =
+  of_sexp_errorf loc "fields %s are all missing (exactly one is needed)"
+    (String.concat ~sep:", " names)
+[@@inline never]
+
+let fields_mutual_exclusion_violation loc names =
+  of_sexp_errorf loc "fields %s are mutually-exclusive"
+    (String.concat ~sep:", " names)
+[@@inline never]
+
+let fields_mutually_exclusive
+      ?on_dup fields ((Fields (loc, _, _) : _ context) as ctx) state =
+  let res, state =
+    traverse (fun (name, parser) ->
```

Comment:
IIUC, `traverse` *chains* the various parser one after the other. Given that we are only using one of them, it feels like we should run then in parallel. It shouldn't make much difference given that they are mutually exclusive but it seems simpler.

Reply:
Looking closer we discovered that there's no applicable notion of parallelism in this context so keeping this unchanged.